exec/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -874,6 +880,46 @@ public void setSafe(int index, BigDecimal value) {
       set(index, value);
     }
 
+    /**
+     * Copies the bulk input into this value vector and extends its capacity if necessary.
+     * @param input bulk input
+     */
+    public <T extends VLBulkEntry> void setSafe(VLBulkInput<T> input) {
+      setSafe(input, null);
+    }
+
+    /**
+     * Copies the bulk input into this value vector and extends its capacity if necessary. The callback
+     * mechanism allows decoration as caller is invoked for each bulk entry.
+     *
+     * @param input bulk input
+     * @param callback a bulk input callback object (optional)
+     */
+    public <T extends VLBulkEntry> void setSafe(VLBulkInput<T> input, VLBulkInput.BulkInputCallback<T> callback) {
```

Comment:
Seems this code is unique to Parquet (in which data is columnar.) Should it be part of the vector itself? Or, should it be in the Parquet-specific vector writers?